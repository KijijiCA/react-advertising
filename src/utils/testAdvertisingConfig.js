export const DIV_ID_FOO = 'div-gpt-ad-foo';
const SLOT_ID_FOO = 666;
const PREBID_SIZES_FOO = [[320, 240]];
const BIDDER_FOO = 'bidder-foo';
const PARAMS_FOO = { foo: 'bar' };

export const DIV_ID_BAR = 'div-gpt-ad-bar';
const SLOT_ID_BAR = [11, 666];
const PREBID_SIZES_BAR = [[320, 240]];
const BIDDER_BAR = 'bidder-bar';
const PARAMS_BAR = { bar: 'foo' };
const COLLAPSE_EMPTY_DIV = [true, true];

const GPT_SIZES = ['fluid', [320, 240]];

const GLOBAL_AD_UNIT_PATH = 'global/ad/unit/path';
const SLOT_AD_UNIT_PATH = 'slot/ad/unit/path';
const USD_TO_EUR_RATE = 2;

export const TWO_SLOT_QUEUE = [
  {
    id: DIV_ID_FOO,
    customEventHandlers: {},
  },
  {
    id: DIV_ID_BAR,
    customEventHandlers: {},
  },
];

export const configWithoutSlots = {
  active: true,
  path: GLOBAL_AD_UNIT_PATH,
  aps: {
    pubID: 'apsid',
    bidTimeout: 2e3,
    deals: true,
  },
  prebid: {
    bidderTimeout: 1500,
    priceGranularity: 'medium',
    bidderSequence: 'random',
  },
  sizeMappings: {
    mobailAndTablet: [
      {
        viewPortSize: [0, 0],
        sizes: [],
      },
      {
        viewPortSize: [320, 700],
        sizes: [
          [300, 250],
          [320, 50],
        ],
      },
      {
        viewPortSize: [1050, 200],
        sizes: [],
      },
    ],
  },
  metaData: {
    usdToEurRate: USD_TO_EUR_RATE,
  },

  targeting: {
    eagt: [666],
    'mt-ab': 'test',
    'mt-ma': ['Poserwagen'],
    'mt-mo': ['Brummstinko', 'Grand Umweltverpestino'],
    'mt-thread': [666],
    'mt-u2': ['00'],
    'mt-u4': true,
  },
  customEvents: {
    collapse: {
      eventMessagePrefix: 'CloseAdvContainer:',
      divIdPrefix: 'div-gpt-ad-',
    },
  },
};
export const config = {
  ...configWithoutSlots,
  slots: [
    {
      id: DIV_ID_FOO,
      targeting: { a: SLOT_ID_FOO },
      sizes: GPT_SIZES,
      prebid: [
        {
          mediaTypes: {
            banner: {
              sizes: PREBID_SIZES_FOO,
            },
          },
          bids: [
            {
              bidder: BIDDER_FOO,
              params: PARAMS_FOO,
            },
          ],
        },
      ],
    },
    {
      id: DIV_ID_BAR,
      path: SLOT_AD_UNIT_PATH,
      targeting: { a: SLOT_ID_BAR },
      sizes: GPT_SIZES,
      sizeMappingName: 'mobailAndTablet',
      collapseEmptyDiv: COLLAPSE_EMPTY_DIV,
      prebid: [
        {
          mediaTypes: {
            banner: {
              sizes: PREBID_SIZES_BAR,
            },
          },
          bids: [
            {
              bidder: BIDDER_BAR,
              params: PARAMS_BAR,
            },
          ],
        },
      ],
    },
  ],
};
