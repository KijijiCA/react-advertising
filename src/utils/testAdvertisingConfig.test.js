import { config } from './testAdvertisingConfig';

describe('The test advertising configuration for unit tests', () =>
  void it('is correct', () => expect(config).toMatchSnapshot()));
