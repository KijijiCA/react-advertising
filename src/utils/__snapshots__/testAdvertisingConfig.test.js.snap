// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The test advertising configuration for unit tests is correct 1`] = `
Object {
  "active": true,
  "customEvents": Object {
    "collapse": Object {
      "divIdPrefix": "div-gpt-ad-",
      "eventMessagePrefix": "CloseAdvContainer:",
    },
  },
  "metaData": Object {
    "usdToEurRate": 2,
  },
  "path": "global/ad/unit/path",
  "prebid": Object {
    "bidderSequence": "random",
    "bidderTimeout": 1500,
    "priceGranularity": "medium",
  },
  "sizeMappings": Object {
    "mobailAndTablet": Array [
      Object {
        "sizes": Array [],
        "viewPortSize": Array [
          0,
          0,
        ],
      },
      Object {
        "sizes": Array [
          Array [
            300,
            250,
          ],
          Array [
            320,
            50,
          ],
        ],
        "viewPortSize": Array [
          320,
          700,
        ],
      },
      Object {
        "sizes": Array [],
        "viewPortSize": Array [
          1050,
          200,
        ],
      },
    ],
  },
  "slots": Array [
    Object {
      "id": "div-gpt-ad-foo",
      "prebid": Array [
        Object {
          "bids": Array [
            Object {
              "bidder": "bidder-foo",
              "params": Object {
                "foo": "bar",
              },
            },
          ],
          "mediaTypes": Object {
            "banner": Object {
              "sizes": Array [
                Array [
                  320,
                  240,
                ],
              ],
            },
          },
        },
      ],
      "sizes": Array [
        "fluid",
        Array [
          320,
          240,
        ],
      ],
      "targeting": Object {
        "a": 666,
      },
    },
    Object {
      "collapseEmptyDiv": Array [
        true,
        true,
      ],
      "id": "div-gpt-ad-bar",
      "path": "slot/ad/unit/path",
      "prebid": Array [
        Object {
          "bids": Array [
            Object {
              "bidder": "bidder-bar",
              "params": Object {
                "bar": "foo",
              },
            },
          ],
          "mediaTypes": Object {
            "banner": Object {
              "sizes": Array [
                Array [
                  320,
                  240,
                ],
              ],
            },
          },
        },
      ],
      "sizeMappingName": "mobailAndTablet",
      "sizes": Array [
        "fluid",
        Array [
          320,
          240,
        ],
      ],
      "targeting": Object {
        "a": Array [
          11,
          666,
        ],
      },
    },
  ],
  "targeting": Object {
    "eagt": Array [
      666,
    ],
    "mt-ab": "test",
    "mt-ma": Array [
      "Poserwagen",
    ],
    "mt-mo": Array [
      "Brummstinko",
      "Grand Umweltverpestino",
    ],
    "mt-thread": Array [
      666,
    ],
    "mt-u2": Array [
      "00",
    ],
    "mt-u4": true,
  },
}
`;
